sdk/core/Azure.Core/src/RequestFailedException.cs

```diff
@@ -258,6 +259,7 @@ internal static bool TryExtractErrorContent(Response response, out ResponseError
                     return false;
                 }
                 error = System.Text.Json.JsonSerializer.Deserialize<ErrorResponse>(content)?.Error;
+                error ??= System.Text.Json.JsonSerializer.Deserialize<ResponseError>(content);
```

Comment:
Did you consider reworking the ResponseError implementation to use the reader approach Krzysztof mentioned?

Reply:
I did, but decided against it.

Reply:
One reason I didn't love the custom parsing approach is that it relies on our custom code to constrain the allowable shape of the json. While this is more flexible, it allows us to more eagerly de-serialize something that is not a ResponseError, but happens to have some of the same property names in the right level of the object hierarchy. I think this approach is safer, and perf is not a big concern for the minority of clients where this path gets taken in the exception path.

Reply:
> One reason I didn't love the custom parsing approach is that it relies on our custom code to constrain the allowable shape of the json.

Can you say more about that?  I don't think I have a good understanding of the objection.

One concern I still have about the current approach is that we're parsing the error in two different ways depending on the code path.  If it goes through the RFE constructor, it will accommodate the outer `error` block, but if a customer uses ResponseError directly, they have to manually remove the `error` block themselves.  If we've decided we will handle both formats, it feels to me like we should be consistent in our approach, or we set people up for mistakes - Heath's sample PR shows [how easy it is to misunderstand](https://github.com/Azure/azure-sdk-for-net/pull/36226#discussion_r1192642540)!

Reply:
Do you think that is the appropriate scope from the perspective of Azure.Core as a library?

The original ask also sounded interested in driving consistency across Azure.Core libraries -- for example, it sounds like Java is targeting having an error type that can deserialize from both formats, and other languages already do.  If we've agreed to align with them, I guess I'm still about confused why we wouldn't implement that functionality here.  Can you help me understand the reason?